File: src/main/java/vubq/warehouse_management/VT_EcoStorage/services/StatisticalService.java
package vubq.warehouse_management.VT_EcoStorage.services;

import vubq.warehouse_management.VT_EcoStorage.dtos.ProductStatisticalDto;
import vubq.warehouse_management.VT_EcoStorage.dtos.responses.ReferenceDataStatisticalResponse;

import java.util.Date;
import java.util.List;

public interface StatisticalService {

    List<ProductStatisticalDto> getStatistical(Date startDate, Date endDate, String warehouseId, String keyword, boolean onlyWithTransaction);

    ReferenceDataStatisticalResponse getReferenceDataStatistical();
}

File: src/main/webapp/src/service/api/statistical-service.ts
import { request } from '../http'

export const StatisticalService = {
  getStatistical(startDate: string, endDate: string, warehouseId: string, keyword: string, onlyWithTransaction: boolean) {
    return request.Get<Service.ResponseResult<any>>('/api/statistical?' + 'startDate=' + startDate + '&endDate=' + endDate + '&warehouseId=' + warehouseId + '&keyword=' + keyword + '&onlyWithTransaction=' + onlyWithTransaction)
  },
  getReferenceData() {
    return request.Get<Service.ResponseResult<any>>('/api/statistical/reference-data')
  },
}
